ui/lib/CreateStream.tsx

```diff
@@ -71,60 +162,247 @@ const CreateStream = (props: CreateStreamProps) => {
     functionName: "approve",
     args: [sm.address, selectedTime * props.amountPerSecond],
   });
-  const { write: approveStream } = useContractWrite(approvalConfig);
+
+  const {
+    data,
+    isLoading,
+    // isSuccess: Make sure transaction has been approved by user (used to get tx hash)
+    isSuccess,
+    isError,
+    write: approveStream,
+  } = useContractWrite(approvalConfig);
+
+  // Then make sure transaction has been processed once it has been approved by user
+  const [txHash, setTxHash] = useState<string | undefined>(undefined);
+  useEffect(() => {
+    if (isSuccess && data?.hash) {
+      setTxHash(data?.hash);
+    }
+  }, [isSuccess, data?.hash]);
+
+  const {
+    error: txError,
+    isSuccess: txSuccess,
+    isLoading: txLoading,
+  } = useWaitForTransaction({
+    hash: txHash as `0x${string}`,
+  });
+
   const createStream = () => {
-    // NOTE: This function should move away from this component
-    sm.create(props.tokenAddress, props.amountPerSecond, props.reasonCode).then(
-      props.registerStream,
-    );
+    props.handleTransactionStatus(true, false, null);
+
+    props
+      .renderReasonCode()
+      .then((reasonString) => {
+        return sm
+          .create(props.tokenAddress, props.amountPerSecond, reasonString)
+          .then((result) => {
+            props.registerStream(result);
+            props.handleTransactionStatus(false, true, null);
+          })
+          .catch((error) => {
+            props.handleTransactionStatus(false, false, error);
+          });
+      })
+      .catch((error) => {
+        props.handleTransactionStatus(false, false, error);
+      });
   };
 
-  return (
-    <div id="CreateStream">
+  // Set transaction amount
+  const transactionAmount = Number(
+    (
+      (selectedTime * props.amountPerSecond) /
+      Math.pow(10, tokenData?.decimals || 0)
+    ).toFixed(Math.min(tokenData?.decimals || 0, 3))
+  );
+
+  // Set card steps logic
+  // Set default selected token to USDC
+  const [selectedToken, setSelectedToken] = useState(
+    "0x7F5c764cBc14f9669B88837ca1490cCa17c31607"
+  );
+
+  type Token = {
+    address: string;
+    symbol: string;
+  };
+  const [tokens, setTokens] = useState<Token[]>([]);
```

Comment:
move this to a prop and pass through in demo app

also, use `@uniswap/token-lists` types here